File: python/organisms/plants/sosnowskys_hogweed.py
from gui.logger import Logger
from organisms.animals import CyberSheep
from organisms.animals.animal import Animal
from organisms.plants.plant import Plant
from organisms.species import Species


class SosnowskysHogweed(Plant):
    def __init__(self, pos):
        super().__init__(pos)
        self.strength = 10
        self.species = Species.SOSNOWSKYS_HOGWEED

    def collision(self, other, attacking, grid):
        if other.species != Species.CYBER_SHEEP:
            grid.set_organism(other.position, None)
            Logger.log("Sosnowsky's hogweed killed " + other.species.name)
            return True
        return False

    def make_action(self, grid):
        # kill all adjacent organisms
        neighbors = grid.get_neighbor_cells(self.position)
        for neighbor in neighbors:
            if neighbor is not None and isinstance(grid.get_organism(neighbor), Animal) and not isinstance(grid.get_organism(neighbor), CyberSheep):
                Logger.log("Sosnowsky's hogweed killed " + grid.get_organism(neighbor).species.name)
                grid.set_organism(neighbor, None)
        super().make_action(grid)

File: java/pl/edu/pg/mygame/organisms/plants/SosnowskysHogweed.java
package pl.edu.pg.mygame.organisms.plants;

import pl.edu.pg.mygame.Grid;
import pl.edu.pg.mygame.World;
import pl.edu.pg.mygame.organisms.Organism;
import pl.edu.pg.mygame.organisms.Species;
import pl.edu.pg.mygame.organisms.animals.Animal;

import java.awt.*;
import java.util.List;

public class SosnowskysHogweed extends Plant {
    public SosnowskysHogweed(Point p) {
        super(p);
        this.species = Species.SOSNOWSKYS_HOGWEED;
        this.strength = 10;
    }

    @Override
    public boolean collision(Organism other, boolean attacking, Grid grid) {
        grid.setOrganism(other.getPosition(), null);
        World.message_panel.showMessage("Sosnowsky's hogweed killed " + other.getSpecies() + "\n");
        return true;
    }

    @Override
    public void makeAction(Grid grid) {
        // kill all adjacent organisms
        List<Point> neighbours = grid.getNeighbourCells(position);
        for (Point neighbour : neighbours) {
            if (neighbour != null && grid.getOrganism(neighbour) instanceof Animal) {
                World.message_panel.showMessage("Belladonna killed " + grid.getOrganism(neighbour).getSpecies() + "\n");
                grid.setOrganism(neighbour, null);
            }
        }
        super.makeAction(grid);
    }
}
